### PR TITLE
Execute integration tests with additional modules

### PR DESCRIPTION
**Pull request by thovid**
* Extracted ComposerApplication bootstrap to be used in tests to include module setup in tests
* Refactored error handling client module
* Added integration test for error handling during upstream request

### CODE REVIEW

File: src/main/java/com/rewedigital/composer/ComposerApplication.java [hunk 1]
```diff
@@ -25,12 +29,25 @@ public static void main(final String[] args) throws LoadingException {
     }
 
     private static Service bootstrapService() {
-        return HttpService
-            .usingAppInit(Initializer::init, "composer")
-            .withEnvVarPrefix("COMPOSER")
-            .withModule(HttpClientModule.create())
-            .withModule(ClientDecoratingModule.create(new ErrorClientDecorator()))
-            .build();
+        return bootstrap(
```
Comment (AndreasKl) [inline src/main/java/com/rewedigital/composer/ComposerApplication.java]: Could you explain the "why" we need this complexity?

File: src/main/java/com/rewedigital/composer/ComposerApplication.java [hunk 2]
```diff
@@ -25,12 +29,25 @@ public static void main(final String[] args) throws LoadingException {
...
+            HttpService::usingAppInit,
+            (b, m) -> b.withModule(m))
+                .withEnvVarPrefix("COMPOSER")
+                .build();
+    }
+
+    static <T> T bootstrap(final BiFunction<AppInit, String, T> init,
```
Comment (AndreasKl) [inline src/main/java/com/rewedigital/composer/ComposerApplication.java]: Can `T` be really anything? Does a type constraint make sense?

Comment (thovid) [reply src/main/java/com/rewedigital/composer/ComposerApplication.java]: Unfortunately, Service Builder and ServiceHelper do not have a common interface

File: src/main/java/com/rewedigital/composer/ComposerApplication.java (hunk 1 above)
Comment (thovid) [reply src/main/java/com/rewedigital/composer/ComposerApplication.java]: This was the simplest thing I could come up with to add the same modules both here and to the ServiceHelper used in our integration tests

**Pull request merged**